File: backend-ngeguidein/app/Models/Destinasi.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Destinasi extends Model
{
    use HasFactory;

    protected $fillable = [
       'nama_tempat',
       'deskripsi',
       'src',
    //    'id_destinasi'
    ];

    public function step_destinasi() {
        return $this->hasMany(Step_destinasi::class);
    }

}

File: tugasbesarflutter/lib/models/Destinasi.dart
// import 'package:tugasbesarflutter/models/StepDestinasi.dart';

class Destinasi {
  final int idDestinasi;
  final String namaTempat;
  final String deskripsi;
  final String src;
  final double latitude;
  final double longitude;
  // final StepDestinasi petunjuk;

  const Destinasi({
    required this.idDestinasi,
    required this.namaTempat,
    required this.deskripsi,
    required this.src,
    required this.latitude,
    required this.longitude,
    // required this.petunjuk,
  });

  factory Destinasi.fromJson(Map<String, dynamic> json) {
    return Destinasi(
        idDestinasi: json['id_destinasi'],
        namaTempat: json['nama_tempat'],
        deskripsi: json['deskripsi'],
        src: json['src'],
        latitude: double.parse(json['latitude']),
        longitude: double.parse(json['longitude']));
  }
}
